myhpi/core/models.py

```diff
@@ -79,12 +88,20 @@ class MinutesList(BasePage):
     ]
     subpage_types = ["Minutes"]
 
-    def get_context(self, request, *args, **kwargs):
-        context = super().get_context(request, *args, **kwargs)
+    def get_visible_minutes(self, request):
         minutes_ids = self.get_children().exact_type(Minutes).values_list("id", flat=True)
+        user_groups = get_user_groups(request.user)
         minutes_list = (
-            Minutes.objects.filter(id__in=minutes_ids, group=self.group).order_by("date").reverse()
+            Minutes.objects.filter(id__in=minutes_ids, group=self.group)
+            .filter(Q(visible_for__in=user_groups) | Q(is_public=True))
+            .order_by("date")
+            .reverse()
```

Comment:
not sure what reverse does, but it is probably more efficient to order by `-date`

Reply:
I don't know about efficient, but possibly more legible